### ISSUE TITLE
Error parsing response metadata

### ISSUE THREAD

**Issue by jonahweissman** [deepgram/deepgram-python-sdk#234]
## What is the current behavior?

When I try to access the request_id field in the response metadata, I get an error.

## Steps to reproduce

This should be a standalone example. 

```
import json
import unittest
from unittest.mock import patch, Mock

from deepgram import DeepgramClient, PrerecordedOptions

class TestDeepgramTranscriber(unittest.IsolatedAsyncioTestCase):
    @patch('httpx.Client.request', new_callable=Mock)
    async def test_transcribe_url_no_retry_example(self, mock_post):
        api_key = 'your_api_key'
        deepgram_client = DeepgramClient(api_key)

        mock_response = Mock()
        mock_response.status_code = 200
        mock_response.raise_for_status = Mock()
        expected_request_id = "ba230c68-47f6-4c68-b03f-33e0f19dedae"
        mock_response.text = json.dumps(
            {
                "metadata": {
                    "channels": 1,
                    "created": "2023-12-30T00:51:08.243Z",
                    "duration": 462.98657,
                    "model_info": {
                        "a375937a-9156-40cb-940d-6a1103040ef1": {
                            "arch": "nova-2",
                            "name": "2-general-nova",
                            "version": "2023-11-14.3290"
                        }
                    },
                    "models": [
                        "a375937a-9156-40cb-940d-6a1103040ef1"
                    ],
                    "request_id": expected_request_id,
                    "sha256": "f5c0cc1f0f17c4922b098db094db3d14a34c7ab97d73b157160e31dbb0c3cee2",
                    "tags": [
                        "174.172.212.100"
                    ],
                    "transaction_key": "deprecated"
                },
                'results': {'channels': [{'alternatives': [{'transcript': 'Hello world'}]}]}
            }
        )

        mock_post.return_value = mock_response

        audio_url = 'https://example.com/audio.mp3'
        source = {'url': audio_url}
        options = PrerecordedOptions(
            punctuate=False,
            model="nhtsa",
            version="v1_dc",
        )
        response = await deepgram_client.listen.asyncprerecorded.v("1").transcribe_url( source, options)
        request_id = response['metadata']['request_id']
        result = response["results"]["channels"][0]["alternatives"][0]["transcript"]

        # Assert that the request was not retried and the result is as expected
        mock_post.assert_called_with(
            'POST',
            'https://api.deepgram.com/v1/listen?model=nhtsa&punctuate=false&version=v1_dc',
            headers={'Accept': 'application/json', 'Authorization': 'Token your_api_key'},
            json={'url': 'https://example.com/audio.mp3'}
        )

        self.assertEqual(result, 'Hello world')
        self.assertEqual(request_id, expected_request_id)
```
Output:
<details>
<pre>
======================================================================
ERROR: test_transcribe_url_no_retry_example (tests.test_deepgram.TestDeepgramTranscriber.test_transcribe_url_no_retry_example)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jonah/.pyenv/versions/3.11.4/lib/python3.11/unittest/async_case.py", line 90, in _callTestMethod
    if self._callMaybeAsync(method) is not None:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jonah/.pyenv/versions/3.11.4/lib/python3.11/unittest/async_case.py", line 112, in _callMaybeAsync
    return self._asyncioRunner.run(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jonah/.pyenv/versions/3.11.4/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jonah/.pyenv/versions/3.11.4/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/jonah/.pyenv/versions/3.11.4/lib/python3.11/unittest/mock.py", line 1392, in patched
    return await func(*newargs, **newkeywargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jonah/dc-vision-zero/Data_Driven_Streets/lambdas/base_image/tests/test_deepgram.py", line 54, in test_transcribe_url_no_retry_example
    request_id = response['metadata']['request_id']
                 ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
  File "/home/jonah/.cache/pypoetry/virtualenvs/data-driven-streets-common-7Blyz5Aa-py3.11/lib/python3.11/site-packages/deepgram/clients/prerecorded/v1
/response.py", line 68, in __getitem__
    _dict["model_info"] = [
                          ^
  File "/home/jonah/.cache/pypoetry/virtualenvs/data-driven-streets-common-7Blyz5Aa-py3.11/lib/python3.11/site-packages/deepgram/clients/prerecorded/v1
/response.py", line 69, in <listcomp>
    ModelInfo.from_dict(model_info) for model_info in _dict["model_info"]
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jonah/.cache/pypoetry/virtualenvs/data-driven-streets-common-7Blyz5Aa-py3.11/lib/python3.11/site-packages/dataclasses_json/api.py", line
70, in from_dict
    return _decode_dataclass(cls, kvs, infer_missing)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jonah/.cache/pypoetry/virtualenvs/data-driven-streets-common-7Blyz5Aa-py3.11/lib/python3.11/site-packages/dataclasses_json/core.py", line
 150, in _decode_dataclass
    kvs = {decode_names.get(k, k): v for k, v in kvs.items()}
                                                 ^^^^^^^^^
AttributeError: 'str' object has no attribute 'items'
</pre>
</details>

## Expected behavior

I would like to be able to get the request_id

## Please tell us about your environment

> We want to make sure the problem isn't specific to your operating system or programming language.
  
- **Operating System/Version:** Pop!_OS 22.04
- **Language:**  Python

## Other information

The problem is that the code is iterating over a dict, which just iterates over the keys.

https://github.com/deepgram/deepgram-python-sdk/blob/d20624ef5594a8ed5c5a567482caca23cccc6b56/deepgram/clients/prerecorded/v1/response.py#L69

One way to make this work would be:
```
 ModelInfo.from_dict(model_info) for _, model_info in _dict["model_info"].items()
```

**Comment by jonahweissman** [comment]
I've been using  `request_id = response['metadata'].request_id` as a workaround, but it'd be nice to see this fixed for other metadata keys.

**Comment by dvonthenen** [comment]
This sounds like a bug. Will take a look at this first thing next week.

Have a happy new year!

**Comment by dvonthenen** [comment]
taking a look at it today

**Comment by dvonthenen** [comment]
There is a PR open to address this issue.

**Comment by dvonthenen** [comment]
Hi @jonahweissman  When `v3.0.2` is released, this should be fixed. In the meantime, if you want to test `main` to verify this fixes your problem.

**Comment by jonahweissman** [comment]
I just re-ran the above unit test `main`, and now I'm seeing an error on a different line:
```
  File "/home/jonah/.cache/pypoetry/virtualenvs/deepgram-test-6cM42gM_-py3.11/lib/python3.11/site-packages/deepgram/clients/prerecorded/v1/response.py"
, line 66, in __getitem__
    _dict["models"] = [str(models) for _, models in _dict["models"].items()]
                                                    ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'list' object has no attribute 'items'
```

**Comment by dvonthenen** [comment]
Oops looks like I went a little too far. Will fix that tomorrow morning.

**Comment by dvonthenen** [comment]
New PR to revert the intented changes
https://github.com/deepgram/deepgram-python-sdk/pull/241

**Comment by dvonthenen** [comment]
That PR got merged and I think we are all good again @jonahweissman

**Comment by dvonthenen** [comment]
This should be available in `v3.0.2`:
https://github.com/deepgram/deepgram-python-sdk/releases/tag/v3.0.2
